### PR TITLE
MOL-530

### PR DESCRIPTION
**Pull request by omBratteng**
Re-writes all database functions to use `AsyncSession`, and functions the creates new entries in the database, uses `flush` instead of `commit`, this allows us to execute a `ROLLBACK` instead of deleting the entry.

### CODE REVIEW

File: mottak-arkiv-service/app/routers/arkivuttrekk.py [hunk 1]
```diff
@@ -64,16 +64,19 @@ async def router_get_by_id(id: int, db: Session = Depends(get_db_session)):
               status_code=status.HTTP_200_OK,
               response_model=Arkivuttrekk,
               summary="Oppdatere felter på et arkivuttrekk")
-async def router_create_arkivuttrekk(id: int, arkivuttrekk: ArkivuttrekkUpdate, db: Session = Depends(get_db_session)):
+async def router_patch_arkivuttrekk(id: int, arkivuttrekk: ArkivuttrekkUpdate, db: AsyncSession = Depends(get_async_db_session)):
```
Comment (kriwal-arkivverket) [inline mottak-arkiv-service/app/routers/arkivuttrekk.py:67]: Litt lang linje?

File: mottak-arkiv-service/app/database/dbo/mottak.py [hunk 2]
```diff
@@ -1,67 +1,75 @@
...
-    arkiv_sluttdato = Column(Date, nullable=False)
-    storrelse = Column(BigInteger(), nullable=False)
-    avtalenummer = Column(String(), nullable=False)
-    opprettet = Column(DateTime(), server_default=func.now(), nullable=False)
-    endret = Column(DateTime(), server_default=func.now(), onupdate=func.current_timestamp(), nullable=False)
+    id: int = Column(Integer(), autoincrement=True, nullable=False, primary_key=True, unique=True)
+    obj_id: uuid.UUID = Column(UUID(as_uuid=True), nullable=False, index=True, unique=True)
+    status: ArkivuttrekkStatus = Column(Enum('Opprettet', 'Under behandling', 'Avvist', 'Sendt til bevaring', 'Lastet ned',
+                                             name='arkivuttrekk_status_type', create_type=True), nullable=False, index=True)
+    type: ArkivuttrekkType = Column(Enum('Noark3', 'Noark5', 'Fagsystem', "SIARD", name='arkivvuttrekk_type_type', create_type=True),
```
Comment (kriwal-arkivverket) [inline mottak-arkiv-service/app/database/dbo/mottak.py:35]: Litt lang linje?

File: mottak-arkiv-service/app/database/dbo/mottak.py [hunk 3]
```diff
@@ -71,40 +79,40 @@ class Invitasjon(Base):
...
-    status = Column(Enum('Startet', 'OK', 'Avbrutt', 'Feilet', name='overforingspakke_status_type', create_type=True),
-                    nullable=False)
-    opprettet = Column(DateTime(), server_default=func.now(), nullable=False)
-    endret = Column(DateTime(), server_default=func.now(), onupdate=func.current_timestamp(), nullable=False)
+    id: int = Column(Integer(), autoincrement=True, nullable=False, primary_key=True, unique=True)
+    invitasjon_id: int = Column(Integer(), ForeignKey('invitasjon.id'), nullable=False, unique=True)
+    tusd_id: str = Column(String(length=60), nullable=False, unique=True, index=True)
+    tusd_objekt_navn: str = Column(String(), nullable=False)
+    storrelse: int = Column(BigInteger(), nullable=False)
+    status: OverforingspakkeStatus = Column(Enum('Startet', 'OK', 'Avbrutt', 'Feilet', name='overforingspakke_status_type', create_type=True),
```
Comment (kriwal-arkivverket) [inline mottak-arkiv-service/app/database/dbo/mottak.py:87]: Litt lang linje?

File: mottak-arkiv-service/app/database/dbo/mottak.py [hunk 4]
```diff
@@ -71,40 +79,40 @@ class Invitasjon(Base):
...
-    target_name = Column(String(), nullable=False)
-    storage_account = Column(String(), nullable=False)
-    container = Column(String(), nullable=False)
-    sas_token_start = Column(DateTime(), nullable=False)
-    sas_token_slutt = Column(DateTime(), nullable=False)
-    opprettet = Column(DateTime(), server_default=func.now(), nullable=False)
-    endret = Column(DateTime(), server_default=func.now(), onupdate=func.current_timestamp(), nullable=False)
+    id: int = Column(Integer(), autoincrement=True, nullable=False, primary_key=True, unique=True)
+    invitasjon_id: int = Column(Integer(), ForeignKey('invitasjon.id'), nullable=False, unique=False)
+    status: ArkivkopiStatus = Column(Enum('Bestilt', 'Startet', 'OK', 'Feilet', name='arkivkopi_status_type', create_type=True),
```
Comment (kriwal-arkivverket) [inline mottak-arkiv-service/app/database/dbo/mottak.py:109]: Litt lang linje?

**Review by kriwal-arkivverket — APPROVED** [review]

**Review by kriwal-arkivverket — APPROVED** [review]

**Comment by kriwal-arkivverket** [comment]
Godkjent, men kan ikke merges før https://github.com/arkivverket/infra-azure/pull/23 er merget

**Comment by omBratteng** [comment]
https://github.com/arkivverket/infra-azure/pull/23 er merget

**Pull request merged**